src/useTimeoutEffect/useTimeoutEffect.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useEffect } from 'react';
+import { useSyncedRef, useHookableRef } from '..';
+
+type TimeoutID = ReturnType<typeof setTimeout> | null;
+
+const cancelTimeout = (id: TimeoutID) => {
+  if (id) {
+    clearTimeout(id);
+  }
+};
+
+/**
+ * Like `setTimeout` but in the form of a react hook.
+ *
+ * @param callback Callback to be called after the timeout.
```

Comment:
It is better to describe that changes in callback do not lead to timeut reset (some developers might expect it).